src/lib/version/version.c

```diff
@@ -58,6 +58,27 @@ enum FIRMWARE_TYPE {
 	FIRMWARE_TYPE_RELEASE = 255
 };
 
+typedef enum {
+	PATCH=0,
+	MINOR=8,
+	MAJOR=16
+} version_type_t;
+
+static void string_to_int(int8_t mag, version_type_t v_type, uint32_t *ver, int start, const char *tag)
+{
+	const char *curr = &tag[start];
+	uint32_t temp_ver = 0;
+
+	while (curr <= &tag[mag])
+	{
+		temp_ver = (temp_ver << 3) + (temp_ver << 1) + (*curr - '0');
```

Comment:
This does not look correct: it assumes hex, but the version is given in decimal, so it should be: `temp_ver * 10 + (*curr - '0');`